### PR TITLE
Removed loincCode and swabType from deviceType

### LINKED ISSUE #5319: Remove `DeviceType.LoincCode` from backend
### Background
We have recently added the new field name `DeviceTestPerformedLoincCode` to store supported disease and loinc code, which makes the standalone fields obsolete and deprecated.

### Action requested
Remove any usage of `DeviceType.LoincCode` from the backend.

### Acceptance Criteria

- LoincCode should be removed from DeviceType model.
- LoincCode should be removed from DeviceType database table.
- Change [TestEventExport](https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java) to return the loinc from the covid disease from the `DeviceTestPerformedLoincCode` relationship

### PR DESCRIPTION
**Pull request by johanna-skylight**
# BACKEND PULL REQUEST

## Related Issue

- Resolves #5319 

## Changes Proposed

- Remove the loincCode, swabType and testPerformed field in the TestResult type
- Delete the TestPerformed type
- Delete all references to the TestPerformed class as it is no longer used
- Update the TestResult data resolvers
- Update unit testing
- Update FHIR converter to use testOrderedLoinc in diagnostic report and testPerformedLoinc in observations
## Additional Information

There will be a following PR with the DB changes

## Testing

_Tested in dev2_
Performed a test and verify that the results works as expected
Verify that the FHIR bundle looks good as well

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java [hunk 1]
```diff
@@ -597,7 +597,12 @@ public String getOrderingProviderPhoneNumber() {
 
   @JsonProperty("Ordered_test_code")
   public String getOrderedTestCode() {
-    return deviceType.map(DeviceType::getLoincCode).orElse(null);
+    // This field is mapped to the testPerformedLoinc but was mistakenly named ordered_test_code
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:600]: ooo this is a helpful note! ty!

**Review by emyl3 — DISMISSED** [review]
LGTM! Tested on dev2! Thanks for all the cleanup 🧹

**Review by BobanL — COMMENTED** [review]
Code looks good so far, I'm going to wait to test it when the conflicts are resolved.

**Review by BobanL — APPROVED** [review]
Tested and changes look good in dev2!

**Review by emyl3 — APPROVED** [review]
LGTM!! Thank you for addressing the merge conflicts 🙌

**Pull request merged**